src/theory/strings/type_enumerator.h

```diff
@@ -125,6 +120,21 @@ class StringEnumeratorLength {
   }
 
   bool isFinished() { return d_curr.isNull(); }
+
+ private:
+  /** The type we are enumerating */
+  TypeNode d_type;
+  /** The cardinality of the alphabet */
+  uint32_t d_cardinality;
```

Comment:
It feels weird to have a cardinality here. Shouldn't this be a property of the type? E.g. sequences of integers have an infinite alphabet.

Reply:
Right, this will be generalized when sequences are supported.  For now, the EnumeratorLength class is guarded by only supporting `String`.  I can try to make more progress on this in this PR, but I wanted to keep the changes small.